docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+
+    def generate_runtime_dockerfile(
+        self,
+        cuda_ver_major: int,
+        cuda_ver_minor: int,
+        repo_ver: str,
+        *,
+        arch: str = "x86_64",
+        nvidia_visible_devices: str = "all",
+        nvidia_driver_capabilities: str = "compute,utility",
+    ) -> str:
+        """
+        Generates a dockerfile for the CUDA runtime image.
+
+        Parameters
+        ----------
+        cuda_ver_major : int
+            The major CUDA version.
+        cuda_ver_minor : int
+            The minor CUDA version.
+        nvidia_visible_devices : str, optional
+            The value to set the nvidia_visible_devices environment variable
+            to, by default "all"
+        nvidia_driver_capabilities : str, optional
+            The value to set the nvidia_driver_capabilities environment
+            variable to, by default "compute,utility"
+
+        Returns
+        -------
+        str
+            The generated dockerfile body
+
+        Raises
+        ------
+        ValueError
+            If the major CUDA version is below 11, the minimum supported
+            version.
+        """
+
+        if cuda_ver_major < 11:
+            raise ValueError(  # pragma: no cover
+                "CUDA version below 11 requested. Only CUDA versions >= 11 "
+                "are supported."
+            )
+        cuda_pkg_version = '"${CUDA_VERSION_MAJOR}-${CUDA_VERSION_MINOR}"'
+        nvidia_req_cuda: str = "cuda>=${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}"
+        body: str = self._generate_initial_lines(repo_ver, arch=arch)
+        body += (
+            "\n"
+            + textwrap.dedent(
+                f"""\
+            ENV NVIDIA_VISIBLE_DEVICES {nvidia_visible_devices}
+            ENV NVIDIA_DRIVER_CAPABILITIES {nvidia_driver_capabilities}
+            ENV CUDA_VERSION_MAJOR {cuda_ver_major}
+            ENV CUDA_VERSION_MINOR {cuda_ver_minor}
+            ENV CUDA_PKG_VERSION {cuda_pkg_version}
+            ENV NVIDIA_REQUIRE_CUDA {nvidia_req_cuda}
+            """
+            ).strip()
+        )
+        body += "\n" + self.generate_install_lines(
+            build_targets=self._runtime_build_targets
+        )
+        return body
+
+    def generate_dev_dockerfile(self) -> str:
+        """
+        Generates a dockerfile for the CUDA Dev image.
+
+        Returns
+        -------
+        str
+            The generated dockerfile body.
+        """
+        body = self.generate_install_lines(build_targets=self._dev_build_targets)
+
+        body = "USER root\n" + body
+
+        version_name = "${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}"
+        body += (
+            "\n"
+            + textwrap.dedent(
+                f"""\
+                ENV CUDAHOSTCXX=x86_64-conda-linux-gnu-c++
+                ENV CUDACXX=/usr/local/cuda-{version_name}/bin/nvcc
+
+                USER $DEFAULT_USER
+                """
+            ).strip()
+        )
+        return body
```

Comment:
This is another case where it seems cleaner and simpler to compose the Dockerfile body as one big multiline string.

```suggestion
        install_lines = self.generate_install_lines(
            build_targets=self._dev_build_targets
        )
        version_name = "${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}"
        
        return textwrap.dedent(f"""
            USER root
            
            {install_lines}
            
            ENV CUDAHOSTCXX=x86_64-conda-linux-gnu-c++
            ENV CUDACXX=/usr/local/cuda-{version_name}/bin/nvcc
            
            USER $DEFAULT_USER
        """).strip()
```